src/lib/swapper/swappers/OsmosisSwapper/endpoints.ts

```diff
@@ -60,75 +54,79 @@ export const osmosisApi: Swapper2Api = {
+
     const sellAssetIsOnOsmosisNetwork = sellAsset.chainId === osmosisChainId
+    const buyAssetIsOnOsmosisNetwork = buyAsset.chainId === osmosisChainId
 
     const sellAssetDenom = symbolDenomMapping[sellAsset.symbol as keyof SymbolDenomMapping]
     const buyAssetDenom = symbolDenomMapping[buyAsset.symbol as keyof SymbolDenomMapping]
+    const nativeAssetDenom = sellAsset.chainId === osmosisChainId ? 'uosmo' : 'uatom'
 
     const adapterManager = getChainAdapterManager()
     const osmosisAdapter = adapterManager.get(osmosisChainId) as osmosis.ChainAdapter | undefined
     const cosmosAdapter = adapterManager.get(cosmosChainId) as cosmos.ChainAdapter | undefined
+    const sellAssetAdapter = adapterManager.get(sellAsset.chainId) as
+      | cosmos.ChainAdapter
+      | osmosis.ChainAdapter
+      | undefined
 
-    const { REACT_APP_OSMOSIS_NODE_URL: osmoUrl } = getConfig()
-
-    if (!cosmosAdapter || !osmosisAdapter) throw new Error('Failed to get adapters')
+    const { REACT_APP_OSMOSIS_NODE_URL: osmoUrl, REACT_APP_COSMOS_NODE_URL: cosmosUrl } =
+      getConfig()
 
-    let sellAddress
+    if (!cosmosAdapter || !osmosisAdapter || !sellAssetAdapter)
+      throw new Error('Failed to get adapters')
 
-    if (sellAssetIsOnOsmosisNetwork) {
-      sellAddress = from
-
-      if (!sellAddress)
-        throw new SwapError('failed to get osmoAddress', {
-          code: SwapErrorType.EXECUTE_TRADE_FAILED,
-        })
-    } else {
+    if (isIbcTransfer) {
       /** If the sell asset is not on the Osmosis network, we need to bridge the
        * asset to the Osmosis network first in order to perform a swap on Osmosis DEX.
        */
-      sellAddress = from
-
-      if (!sellAddress) throw new Error('Failed to get address')
 
       const transfer = {
-        sender: sellAddress,
+        sender: from,
         receiver: receiveAddress,
         amount: sellAmountBeforeFeesCryptoBaseUnit,
       }
 
-      const responseAccount = await cosmosAdapter.getAccount(sellAddress)
-      const ibcAccountNumber = parseInt(responseAccount.chainSpecific.accountNumber || '0')
+      const responseAccount = await sellAssetAdapter.getAccount(from)
+      const ibcAccountNumber = responseAccount.chainSpecific.accountNumber || '0'
 
       const sequence = responseAccount.chainSpecific.sequence || '0'
 
       const getFeeDataInput: Partial<GetFeeDataInput<OsmosisSupportedChainId>> = {}
-      const ibcFromCosmosFeeData = await cosmosAdapter.getFeeData(getFeeDataInput)
+      const sellAssetFeeData = await sellAssetAdapter.getFeeData(getFeeDataInput)
 
       const unsignedTx = await buildPerformIbcTransferUnsignedTx({
         input: transfer,
-        adapter: cosmosAdapter,
-        blockBaseUrl: osmoUrl,
-        denom: 'uatom',
-        sourceChannel: COSMO_OSMO_CHANNEL,
-        feeAmount: ibcFromCosmosFeeData.fast.txFee,
+        adapter: sellAssetAdapter,
+        // The blockheight of the destination chain for the IBC transfer
```

Comment:
this comment seems out of place 😅

Reply:
eyy nice catch, needs a little rewordy cea29eec3f98b84b68094d94294411085ec8da27